src/QsCompiler/Transformations/TypeResolutionCombination.cs

```diff
@@ -0,0 +1,389 @@
+﻿﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
+
+namespace Microsoft.Quantum.QsCompiler
+{
+    using ExpressionKind = QsExpressionKind<TypedExpression, Identifier, ResolvedType>;
+    using ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+    // Type Parameters are frequently referenced by the callable of the type parameter followed by the name of the specific type parameter.
+    using TypeParameterName = Tuple<QsQualifiedName, NonNullable<string>>;
+    using TypeParameterResolutions = ImmutableDictionary</*TypeParameterName*/ Tuple<QsQualifiedName, NonNullable<string>>, ResolvedType>;
+
+    /// <summary>
+    /// Combines a series of type parameter resolution dictionaries, IndependentResolutionDictionaries,
+    /// into one resolution dictionary, CombinedResolutionDictionary, containing the ultimate type
+    /// resolutions for all the type parameters found in the dictionaries. Validation is done on the
+    /// resolutions, which can be checked through the IsValid flag.
+    /// </summary>
+    public class TypeResolutionCombination
+    {
+        // Static Members
+
+        /// <summary>
+        /// Checks if the given type parameter directly resolves to itself. If so, the
+        /// resolution may be overridden without conflict.
```

Comment:
The second sentence seems irrelevant for what this method does, and I suggest to remove it.